src/components/PathwayExercise/ExerciseContent/index.js

```diff
@@ -0,0 +1,304 @@
+import React, { useEffect, useState, useMemo, useRef } from "react";
+import { useSelector } from "react-redux";
+import { METHODS } from "../../../services/api";
+import axios from "axios";
+import get from "lodash/get";
+import YouTube from "react-youtube";
+import DOMPurify from "dompurify";
+import { useParams } from "react-router-dom";
+import useMediaQuery from "@mui/material/useMediaQuery";
+import { breakpoints } from "../../../theme/constant";
+import CircleIcon from "@mui/icons-material/Circle";
+import {
+  TableRow,
+  TableHead,
+  TableCell,
+  TableBody,
+  Table,
+  TableContainer,
+} from "@mui/material";
+
+// import HiddenContent from "../HiddenContent";
+import { versionCode } from "../../../constant";
+
+import useStyles from "../styles";
+
+import { Container, Box, Typography, Button, Grid } from "@mui/material";
+
+function getMarkdown(code, lang) {
+  let l = lang == "python" ? "py" : "js";
+  return `~~~${l}
+${code}
+~~~`;
+}
+
+const createVisulizeURL = (code, lang, mode) => {
+  // only support two languages for now
+  let l = lang == "python" ? "2" : "js";
+  let replacedCode = code && code.replace(/<br>/g, "\n");
+  let visualizerCode = replacedCode.replace(/&emsp;/g, " ");
+  let url = `http://pythontutor.com/visualize.html#code=${encodeURIComponent(
+    visualizerCode
+  )
+    .replace(/%2C|%2F/g, decodeURIComponent)
+    .replace(/\(/g, "%28")
+    .replace(
+      /\)/g,
+      "%29"
+    )}&cumulative=false&curInstr=0&heapPrimitives=nevernest&mode=${mode}&origin=opt-frontend.js&py=${l}&rawInputLstJSON=%5B%5D&textReferences=false`;
+  return url;
+};
+
+const headingVarients = {
+  1: (data) => (
+    <Typography
+      variant="h6"
+      className="heading"
+      dangerouslySetInnerHTML={{ __html: data }}
+    ></Typography>
+  ),
+  2: (data) => (
+    <h2 className="heading" dangerouslySetInnerHTML={{ __html: data }}></h2>
+  ),
+  3: (data) => (
+    <h3 className="heading" dangerouslySetInnerHTML={{ __html: data }}></h3>
+  ),
+  4: (data) => (
+    <h4 className="heading" dangerouslySetInnerHTML={{ __html: data }}></h4>
+  ),
+  5: (data) => (
+    <h5 className="heading" dangerouslySetInnerHTML={{ __html: data }}></h5>
+  ),
+  6: (data) => (
+    <h6 className="heading" dangerouslySetInnerHTML={{ __html: data }}></h6>
+  ),
+};
```

Comment:
Why does `headingVarients[1]` have a variant of `h6`, and why is it the only one that uses `Typography`?  Remember you could use Typography for each and set a `component` property of 'h_' for each `Typography` (https://mui.com/material-ui/api/typography/#props).  If it needs to be distinctive for some reason, you can still factor out the commonality of the Components like this:

```suggestion
const headingVarients = {};
[Typography, 'h2', 'h3', 'h4', 'h5', 'h6'].forEach(
  (Name, index) =>
    (headingVarients[index + 1] = (data) => (
      <Name
        className="heading"
        dangerouslySetInnerHTML={{ __html: data }}
        {...(index === 0 ? { component: 'h1', variant: 'h6' } : {})}
      />
    ))
);
```

Reply:
No, actually somehow I forget to replace all and didn't even see it. I'll make the changes.